### ISSUE TITLE
README cleanup - shorter and simpler/better "visually" instructions

### ISSUE THREAD

**Issue by ventz** [DaveGut/TP-Link-SmartThings#16]
Hi,

I've been using your device handler and smartapp  since you first published it (~an year ago). It works perfectly, and it's absolutely great. Great job on the coding/development!

One thing that I think can be improved is the README. Everyone of my friends with a tp-link plug/smartthings that I have told about this project has said the same thing -- they have no idea what's happening from the README, and it feels like multiple solutions are mixed into one README. I suspect that's due to the cloud vs hub and "nodejs" mixture. Eventually they all figure it out, but I think this might be an area where things can be improved.

You already have some *fantastic* documentation (ex: https://github.com/DaveGut/TP-Link-SmartThings/blob/master/Documentation/Cloud%20TP-Link%20to%20SmartThing%20Add%20Device.pdf)

I think literally just extracting it into a more simple form for your main page README (maybe a table of contents/something that can be easily followed) will be helpful.

Maybe something like this style, with a table of contents, and more "visual"/rather than code blocks?
https://github.com/Martinsos/edlib

**Comment by DaveGut** [comment]
See the README.md on the first page.  It is the current installation process.  Also, I do not have a lot of time for creating the detailed instruction.  And I will always assume that the user can navigate the IDE, Smart Apps, and other items.

I am going to update the Service Manager to simplify and make more intuitive.  That will help some (I created the current at the urging of a super-user and it is not as intuitive as I would like).

dave

**Comment by ventz** [comment]
I think some of the confusion might be between the:

1.) TP-Link / Kasa Smart Home Device SmartThings Integration
vs
2.) Installation Prequisites

Unless you have used the integration, it's not obvious that those are 3 different "install options/setup scenarios"


Also, the "a/b/c" from the first do not align with the "a/b/c" from the second -- and if you read it through, it looks like something new. (maybe also eliminating `a.) A SmartThing Hub` so they can at least align?


From the main install section:
> Installing the code into the IDE.

^ I think breaking this up into sections of "major" actions, with #'s for the steps within each and making it a bit more "visual" might help.

If you are willing to take pull requests, I am more than willing to do it.

**Comment by DaveGut** [comment]
No.  I am going to update the Service Manager, documentation, then throw away my hub.

**Comment by ventz** [comment]
Throw away your hub? :)

**Comment by DaveGut** [comment]
I have completed all steps to convert to Hubitat except throw away the hub.  I really will not throw it away, just power off until needed to correct issues.  My goals on this final version are

- backward compatibility to 3.5 for both Service Manager to Device Handler and vica-vesa,
- intuitive Service Manager with annotations to explain what is next at each step,
- reduced functionality from current - some functions are not necessary and are causing issues,
- solid external instructions for installing and adding devices.

**Comment by ventz** [comment]
Ah.

I have to say, I am getting pretty frustrated with SmartThings. Everything is "half-broken", the UI is a disgusting mess (and that's from someone that spends all day on the CLI), and it seems like the community is the driver (vs Samsung).

I don't want to spam here, but would love to hear your view of Hubitat. I was looking at OpenHAB a while back, and after that Home Assistant ...both lacked in their own way.

**Comment by DaveGut** [comment]
Hubitat is a hobby platform.  It takes a lot of maintenance and work to get things up and running and the UI is still limited.  Big advantage (over SmartThings) is that the total environment executes from the local Hub.  You only go through the hubitat cloud when you are away from home.

Note:  biggest problem with SmartThings has been the growing SmartThings cloud.  They had a erroneous business model in not charging for cloud access from the beginning.  As the user base grew, revenues did not grow, so money for a growing cloud was not available.  (Opinion).

I am leaving SmartThings mainly because of the future architecture changes where Device Handlers and Smart Apps will have to be hosted by the Author.  When that happens, I would be dead in the water.

**Comment by ventz** [comment]
> I am leaving SmartThings mainly because of the future architecture changes where Device Handlers and Smart Apps will have to be hosted by the Author. When that happens, I would be dead in the water.

You can't keep hosting them on github?

**Comment by DaveGut** [comment]
I am 70 years old,  I am getting too old to learn new things.

**Comment by DaveGut** [comment]
Besides, the hosting is on a cloud device like AWS that will actually execute the code.

**Comment by ventz** [comment]
> I am 70 years old, I am getting too old to learn new things.

Yet you managed to have one of the first, and then (and still) one of the best TP-Link integrations for SmartThings :)

**Comment by sunnyghai** [comment]
I agree with @ventz , you've done a great job.  I hope you don't abandon ST.

**Comment by uSlackr** [comment]
Don’t sell yourself short.  You’ve learned a lot over the past couple years; you have a knack for it and turning into interesting things. (Thanks for that).

Now, all those other things are good reasons for moving beyond ST.
I’ll have to play with Hubistat and maybe learn something new myself (I’m only 59 1/2 😊 )


\\Greg

________________________________

**Comment by DaveGut** [comment]
I just updated the device handlers, smart application, read.me, and documentation for the integration.  I divided the tasks up so that knowledgeable persons can skip through the common steps while still providing details for the novice.

Dave

**Comment by sunnyghai** [comment]
Dave,

I will check it out.  If I update the dh and app, will I have to reinstall the devices?


On Feb 21, 2019, at 6:02 PM, David Gutheinz <notifications@github.com> wrote:

I just updated the device handlers, smart application, read.me, and documentation for the integration. I divided the tasks up so that knowledgeable persons can skip through the common steps while still providing details for the novice.

Dave
